app/models/user.rb

```diff
@@ -5,6 +5,11 @@ class User < ApplicationRecord
          :recoverable, :rememberable, :validatable, :timeoutable, :trackable
   has_many :favorites, dependent: :destroy, inverse_of: :user
   has_many :todaysmenus, dependent: :destroy
+  has_many :stocks, dependent: :destroy
+  # userscontrollerのupdate_profileアクションでのみvalidationを実行したいが、実行されない
+  validates :nickname, presence: true, on: :update_profile
```

Comment:
onオプションは更新や作成時しか指定できないので、定義したアクション毎にはできないです。
https://qiita.com/tmzkysk/items/a0c874715ba38eb23350

この場合は下記のようにすれば更新時には実行されます。
```suggestion
  validates :nickname, presence: true, on: :update
```